src/vos/vos_common.c

```diff
@@ -457,7 +457,7 @@ vos_mod_init(void)
 	if (vos_start_epoch == DAOS_EPOCH_MAX)
 		vos_start_epoch = d_hlc_get();
 
-	rc = vos_pool_settings_init();
+	rc = vos_pool_settings_init(bio_nvme_configured(SMD_DEV_TYPE_META));
```

Comment:
As I mentioned in prior comment, bio_nvme_configured() won't work at this moment. (when vos_mod_init() is called, the bio_nvme_init() isn't called yet). Please be aware that the engine mode is different from standalone mode, we don't call vos_self_init() for engine mode.

Reply:
how do you suggest I fix the situation, can't pass in the bool to vos_mod_init as it's a templated function

Reply:
I think there are two ways to fix it:
1. As you proposed, move the bio_nvme_init() before dss_module_init_all();
2. Move vos_pool_settings_init() out from vos_mod_init() and put it after bio_nvme_init();

Option 1 looks cleaner to me.

Reply:
I'm made the change #1, Done